apps/backend/src/lambdas/parseTDI.ts

```diff
@@ -0,0 +1,114 @@
+import { BadRequestException } from '@nestjs/common';
+import { NestFactory } from '@nestjs/core';
+import { Context } from 'aws-lambda';
+import { AppModule } from '../app.module';
+import { AppLogger } from '../common/logger.service';
+import { S3ManagerService } from '../s3-manager/s3-manager.service';
+import { TDI17Details, TDI34Details, generateEnum } from '../tdi-files';
+
+export const handler = async (
+  event?: { filePath: string; type: string },
+  context?: Context,
+) => {
+  const app = await NestFactory.createApplicationContext(AppModule);
+  const appLogger = app.get(AppLogger);
+  const s3manager = app.get(S3ManagerService);
+
+  appLogger.log({ event });
+  appLogger.log({ context });
+
+  const filetype = event?.type ?? process.env.TDI_TYPE;
+
+  if (!filetype) {
+    throw new BadRequestException();
+  }
+
+  const TDI_FILE = generateEnum(filetype);
+
+  const filepath = event?.filePath ?? TDI_FILE.LOCALSTACK_PATH;
+
+  if (!filepath) {
+    throw new BadRequestException();
+  }
+  await findAndParseFile(TDI_FILE, s3manager, appLogger, filepath);
+};
+
+const findAndParseFile = async (
+  TDI_FILE: {
+    BUCKET: string;
+    OUTPUT_PATH: string;
+    JSON_FILENAME: string;
+    CONTENT_TYPE: string;
+    TYPE: string;
+  },
+  s3manager: S3ManagerService,
+  appLogger: AppLogger,
+  filepath: string,
+) => {
+  try {
+    appLogger.log(`...start ${TDI_FILE.TYPE} Parsing`);
+
+    const contents = await s3manager.getContents(TDI_FILE.BUCKET, filepath);
+
+    const record = contents?.Body?.toString();
+    const output = record && parseTDI(record, TDI_FILE);
+
+    appLogger.log(output);
+
+    await uploadParsedTDI(TDI_FILE, s3manager, output, appLogger);
+  } catch (e) {
+    appLogger.error(e);
```

Comment:
we should exit the handler here gracefully.

Reply:
or pass the error back to the root fn and exit there

Reply:
It is also okay to not handle it because we will see the error in the console, but more logs and structure will help debugging in the future